gui/test/ruler-tests.cc

```diff
@@ -0,0 +1,370 @@
+#define BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
+namespace utf = boost::unit_test;
+
+#include "../src/ruler.hh"
```

Comment:
Doing `..` in a `#include` makes it harder to change the directory structure of the archive (and it breaks encapsulation).

Prefer doing `target_include_directories(src)` in the `CMakeLists.txt`